loadtest/control/simulcontroller/controller.go

```diff
@@ -56,12 +56,18 @@ func New(id int, user user.User, config *Config, status chan<- control.UserStatu
 // It keeps on doing it until Stop() is invoked.
 // This is also a blocking function, so it is recommended to invoke it
 // inside a goroutine.
-func (c *SimulController) Run() {
+func (c *SimulController) Run(started chan struct{}) {
 	if c.user == nil {
 		c.sendFailStatus("controller was not initialized")
+		close(started)
 		return
 	}
 
+	c.disconnectChan = make(chan struct{})
+	c.stopChan = make(chan struct{})
+	c.stoppedChan = make(chan struct{})
```

Comment:
Would moving these into the stopping/cleanup process remove the need for this `started` chan?

Reply:
Assuming that you are thinking of removing these into e.g. `Stop()` method. So, my answer is yes, but in that case we will leave these channels open every time we stop the controller.

Reply:
They were opened on controller creation so the idea would be to leave them as found since we assume a controller could be re-used. 
I think `disconnectChan` can be safely re-initialized in `Run()` `defer` block. The rest can probably be done in `Stop()`.

Reply:
Sure, btw since we don't close the disconnetChan we don't need to re-initialize it.